### PR TITLE
Basic support for extension bundles

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
No more dependency on .NET Core!!! Instead of running `func extensions install` in every project, users just have to specify a default bundle version in their "host.json" file and it will pull down the right binaries for them.

This PR will only affect new projects going forward. In the future, we might consider prompting users to "convert" their project from `func extensions install` to bundles, but I'd rather let this bake a bit before we do that.

There's still some pending open questions being discussed offline. Tracking that here: https://github.com/Microsoft/vscode-azurefunctions/issues/1203 https://github.com/Microsoft/vscode-azurefunctions/issues/1202

### CODE REVIEW

**Comment by EricJizbaMSFT** [comment]
ping @StephenWeatherford @PrashanthCorp got some work that builds off this PR and would like it reviewed ASAP

File: src/commands/createFunction/FunctionCreateStepBase.ts [hunk 1]
```diff
@@ -65,6 +71,23 @@ export abstract class FunctionCreateStepBase<T extends IFunctionWizardContext> e
...
 
+    public async verifyExtensionBundle(wizardContext: T): Promise<void> {
+        const hostFilePath: string = path.join(wizardContext.projectPath, hostFileName);
+        try {
+            const hostJson: IHostJson = <IHostJson>await fse.readJSON(hostFilePath);
+            if (!hostJson.extensionBundle) {
+                // https://github.com/Microsoft/vscode-azurefunctions/issues/1202
+                hostJson.extensionBundle = {
+                    id: 'Microsoft.Azure.Functions.ExtensionBundle',
+                    version: '[1.*, 2.0.0)'
```
[the remark this answers is not in the document]
Comment (EricJizbaMSFT) [reply src/commands/createFunction/FunctionCreateStepBase.ts]: See linked issue a few lines up

[the remark this answers is not in the document]
Comment (PrashanthCorp) [reply src/commands/createFunction/FunctionCreateStepBase.ts]: I was writing in the context of #1202. If they bump up the version often, does it make sense to ask an API endpoint to poll for the supported versions?

[the remark this answers is not in the document]
Comment (EricJizbaMSFT) [reply src/commands/createFunction/FunctionCreateStepBase.ts]: That issue _is_ about getting it from an api. Specifically this one:
https://functionscdn.azureedge.net/public/cli-feed-v3.json

[the remark this answers is not in the document]
Comment (nturinski) [reply src/commands/createFunction/FunctionCreateStepBase.ts]: Well ideally the cli-feed will provider this property anyway, right?

File: src/commands/initProjectForVSCode/InitVSCodeStep/PythonInitVSCodeStep.ts [hunk 2]
```diff
@@ -38,31 +40,35 @@ export class PythonInitVSCodeStep extends ScriptInitVSCodeStep {
...
+        const pipInstallLabel: string = 'pipInstall';
+        const dependsOn: string | undefined = this.requiresFuncExtensionsInstall ? extInstallCommand : this._venvName ? pipInstallLabel : undefined;
         const tasks: TaskDefinition[] = [
             {
                 type: func,
                 command: hostStartCommand,
                 problemMatcher: funcWatchProblemMatcher,
                 isBackground: true,
-                dependsOn: extInstallTaskName
+                dependsOn
```
Comment (nturinski) [inline src/commands/initProjectForVSCode/InitVSCodeStep/PythonInitVSCodeStep.ts]: If there is an `extInstallCommand` the dependsOn property would just have the venv name, wouldn't it?

File: test/validateProject.ts [hunk 3]
```diff
@@ -196,8 +194,7 @@ export function getPowerShellValidateOptions(): IValidateProjectOptions {
         expectedSettings: {
             projectLanguage: ProjectLanguage.PowerShell,
             projectRuntime: ProjectRuntime.v2,
-            deploySubpath: '.',
-            preDeployTask: 'func: extensions install'
```
Comment (nturinski) [inline test/validateProject.ts]: 🎉

File: src/commands/initProjectForVSCode/InitVSCodeStep/PythonInitVSCodeStep.ts (hunk 2 above)
Comment (EricJizbaMSFT) [reply src/commands/initProjectForVSCode/InitVSCodeStep/PythonInitVSCodeStep.ts]: So I don't know about you, but my diff gets cut off at the edge. It looks like this:
Screen Shot 2019-04-22 at 3 38 48 PM

But if you scroll over, I feel like the rest of the line answers your question:
Screen Shot 2019-04-22 at 3 38 55 PM

dependsOn will never evaluate to the venvName. Instead it either evaluates to `pipInstallLabel` or undefined

**Comment by daxian-dbw** [comment]
Does this PR really remove the dependency of dotnet SDK? What the Azure Function VSCode extension do for the `function.proj` files used in a C# script function? I think dotnet SDK is needed to restore `function.proj` file to pull down the dependent nuget packages declared in it.

**Comment by EricJizbaMSFT** [comment]
This only affects new projects. When you say `function.proj` do you mean `extensions.csproj`? If so, that file will no longer be written down in new projects. You're welcome to try this out in the latest build from master as described here: https://aka.ms/azcodeInsiders

There will also be more documentation on this feature coming in the near future.

**Pull request merged**